crates/biome_analyze/CONTRIBUTING.md

```diff
@@ -103,54 +105,154 @@ New rules **must** be placed inside the `nursery` group. This group is meant as
+   # $ just new-json-assistrule useMyRuleName
    ```
-   The script will generate a bunch of files for the _JavaScript_ language, inside the `biome_js_analyze` crate.
-   Among the other files, you'll find a file called `use_my_new_rule_name.rs` inside the `biome_js_analyze/lib/src/lint/nursery` folder. You'll implement your rule in this file.
+   The script `just new-js-lintrule` script will generate a bunch of files for the _JavaScript_ language inside the `biome_js_analyze` crate.
+   Among the other files, you'll find a file called `use_my_rule_name.rs` inside the `biome_js_analyze/lib/src/lint/nursery` folder. You'll implement your rule in this file.
 
-   If you want to create a _CSS_ lint rule, run this script. It will generate a new lint rule in `biome_css_analyze`
+2. Let's have a look at the generated code in  `use_my_rule_name.rs`:
 
-   ```shell
-   just new-css-lintrule useMyRuleName
+
+   ```rust
+   ...
+   impl Rule for UseMyRuleName {
+       type Query = Ast<JsIdentifierBinding>;
+       type State = ();
+       type Signals = Option<Self::State>;
+       type Options = ();
+
+       fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+           ...
+       }
+
+       fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+           ...
+       }
+   }
    ```
 
-1. The `Ast` query type allows you to query the CST of a program.
-1. The `State` type doesn't have to be used, so it can be considered optional. However, it has to be defined as `type State = ()`.
-1. Implement the `run` function:
+   We can observe a few details from this snippet:
+
+   - The **`Options`** type can be used to define [additional options](#rule-options) for your rule:
+
+     ```rust
+     type Options = ();
+     ```
+
+     Use `()` if your rule does not have additional options.
+
+   - The **`Query`** type defines the entities for which your your rule's `UseMyRuleName::run` function will be invoked:
+     ```rust
+     type Query = Ast<JsIdentifierBinding>;
+     ```
+
+     _→ The `Ast<>` query type for example allows you to query the AST/CST of a program._
+
+     _→ For more advanced use cases it is also possible to define [custom query types](#custom-syntax-tree-visitors)._
+
+   - The **`run`** function will be invoked for each match of `Query`.
+     It should return either `Some` to report a diagnostic for that match, or `None`.
+
+     _→ It is also possible to report multiple diagnostics for a single `Query` match._
+     _See [Multiple Signals](#multiple-signals) for instructions._
+
+     ```rust
+     type Signals = Option<Self::State>;
+
+     fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+       let matched_node = ctx.query();
+       ...
+     }
+     ```
+
+   - The **`diagnostic`** function will be invoked for each occurence returned by `run`, and turns these occurences into `RuleDiagnostic` instances that define the message(s) reported to the user.
```

Comment:
```suggestion
   - The **`diagnostic`** function will be invoked for each signal returned by `run`, and these signals will be turned into `RuleDiagnostic` instances that define the message(s) reported to the user.
```

Saying "occurrence" without explaining what "occurrence" is can be confusing. Since we are talking about signals, we should stick with it.

Reply:
I agree, though we should really explain "signal" then, too, because, by itself, that's an even more generic term.

I'd argue for using the active voice here, because emitting the `RuleDiagnostic` instances is exactly what `diagnostic` does. Using the passive voice IMO makes it sound like the `signal => RuleDiagnostic` conversion is done somewhere else.

```suggestion
   - The **`diagnostic`** function will be invoked for each signal returned by `run`, and turns these signal into `RuleDiagnostic` instances that define the message(s) reported to the user.
```